### PR TITLE
docs: add release procedure

### PR DESCRIPTION
**Pull request by JoepVanlier**
Adds release procedures.

### CODE REVIEW

File: release.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Pylake release procedure
+
+## Preflight check
+- Update the changelog with the date.
```
Comment (dean0x7d) [inline release.md:4]: Leave a blank line after titles.

File: release.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+- Bump the version number in `__about__.py`.
+- Check whether any dependencies have changed.
+
+## Releasing
+### Pypi
+- Perform the release procedure on GH (just follow the instructions on "create a new release").
```
Comment (dean0x7d) [inline release.md:10]: I think that GitHub is short enough :)

File: release.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Pylake release procedure
+
+## Preflight check
+- Update the changelog with the date.
+- Bump the version number in `__about__.py`.
+- Check whether any dependencies have changed.
```
Comment (dean0x7d) [inline release.md:6]: Add a point to review the changelog, make sure everything is clear and informative for users.

File: release.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+## Releasing
+### Pypi
+- Perform the release procedure on GH (just follow the instructions on "create a new release").
+- Include the relevant changelog entries.
+- Don't forget to check "This is a pre-release".
+
+### Conda Forge
+- Go to pypi and go to the version you just released.
```
Comment (dean0x7d) [inline release.md:15]:
```suggestion
- Go to the PyPI page of the newly released version: https://pypi.org/project/lumicks.pylake/
```

File: release.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+- Under download files, check the sha256 hash of the tarball. You will need this for releasing on Conda Forge.
+- Use our Conda feedstock fork, set the upstream to the conda-forge master. Rebase the fork to the upstream master.
+- Create a new branch on our feedstock for your release.
```
Comment (dean0x7d) [inline release.md:18]:
```suggestion
- Use our [conda-forge feedstock fork](https://github.com/lumicks/lumicks.pylake-feedstock) and create a new branch based on `upstream/master`.
```

File: release.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+- Update the version in the recipe/meta.yaml if the version number changed. If the build number is not zero, set it to zero. Also update the sha256 hash.
```
Comment (dean0x7d) [inline release.md:19]:
```suggestion
- Update the version in the `recipe/meta.yaml` if the version number changed. If the build number is not zero, set it to zero. Also update the sha256 hash.
```

File: release.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+- If it's a re-release because some dependencies changed, increment the build number but do not change the version.
+See https://conda-forge.org/docs/maintainer/updating_pkgs.html for more info.
+- Make a PR on the upstream conda forge repository. Wait for the tests to complete and pull in the changes.
```
Comment (dean0x7d) [inline release.md:22]:
```suggestion
- Make a PR on the [upstream conda-forge repository](https://github.com/conda-forge/lumicks.pylake-feedstock). Wait for the tests to complete and pull in the changes.
```

**Review by dean0x7d — COMMENTED** [review]
Links are good :)

File: release.md (hunk 1 above)
Comment (JoepVanlier) [reply release.md]: 👍

File: release.md (hunk 2 above)
Comment (JoepVanlier) [reply release.md]: Yes :D

File: release.md (hunk 3 above)
Comment (JoepVanlier) [reply release.md:8]: Done.

File: release.md (hunk 4 above)
Comment (JoepVanlier) [reply release.md]: Changed!

File: release.md (hunk 5 above)
Comment (JoepVanlier) [reply release.md]: Changed!

File: release.md (hunk 6 above)
Comment (JoepVanlier) [reply release.md]: Good catch.

File: release.md (hunk 7 above)
Comment (JoepVanlier) [reply release.md]: 👍

**Review by dean0x7d — APPROVED** [review]
Cool!

**Pull request merged**